app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/AdminDetailsActivity.java

```diff
@@ -0,0 +1,138 @@
+        int totalTicketNumber = 100;
+        int totalSaleNumber = tickets.size();
+        String totalSaleString = "Total Sale : " + totalSaleNumber + "/" + totalTicketNumber;
+        totalSaleView.setText(totalSaleString);
+
+        //Set FAB to scanner
+        floatingScanner.setOnClickListener(view -> scanning());
+        // Set search function
+        setupSearchView();
+
+    }
+
+    public boolean onCreateOptionsMenu(Menu menu) {
+        super.onCreateOptionsMenu(menu);
+        getMenuInflater().inflate(R.menu.menu_activity_admin, menu);
+        menuItemScanView = menu.findItem(R.id.action_refresh);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int i = item.getItemId();
+        if (i == R.id.action_refresh) {
+            //TODO:send event_id to backend and receive really ticket data,following just use dummy ticketdata
+            tickets = TicketsController.getrefreshTickets();
+            mAdapter = new TicketListAdapter(tickets, this);
+            listView.setAdapter(mAdapter);
+            mAdapter.notifyDataSetChanged();
+        }
+        return true;
+    }
+
+    private void scanning() {
+        startActivity(new Intent(this, TicketScanActivity.class));
+        //TODO:send current eventId to TicketScanActivity
+    }
+
+    private void setupSearchView() {
+        searchView.setSubmitButtonEnabled(true);
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            public boolean onQueryTextSubmit(String query) {
+                if (TextUtils.isEmpty(query)) {
+                    Toast.makeText(AdminDetailsActivity.this, "please text something", Toast.LENGTH_SHORT).show();
+                    listView.setAdapter(mAdapter);
+                } else {
+                    findList.clear();
+                    for (int i = 0; i < tickets.size(); i++) {
+                        AdminTicket findticket = tickets.get(i);
+                        if (findticket.getName().toLowerCase().equals(query.toLowerCase())) {
+                            findList.add(findticket);
+                            break;
+                        }
+                    }
+                    if (findList.size() == 0) {
+                        Toast.makeText(AdminDetailsActivity.this, "The name is not in the list", Toast.LENGTH_SHORT).show();
+                    } else {
+                        Toast.makeText(AdminDetailsActivity.this, "Search successfully", Toast.LENGTH_SHORT).show();
+                        findAdapter = new TicketListAdapter(findList, AdminDetailsActivity.this);
+                        listView.setAdapter(findAdapter);
+                    }
+                }
+                return true;
+            }
+
+            public boolean onQueryTextChange(String newText) {
+                if (TextUtils.isEmpty(newText)) {
```

Comment:
onQueryTextSubmit and onQueryTextChange have a lot of code in common. Can we maybe extract this as a method which both methods can use?
If you have statements which should only be executed for one method you can do something like:

private void handleQuery(boolean isSubmission){
    ....
    if (isSubmission) Utils.showToast(...)
    ....
}

Reply:
I find that onQueryTextSubmit and onQueryTextChange  look similar at first glance. But they have much difference in a lot of detail? I may not able to extract them in a method. Others are finished. I add function to search by Id